src/redirects.js

```diff
@@ -0,0 +1,50 @@
+const getDestination = (ctx, AuthUser, redirectDestination) => {
+  if (typeof redirectDestination === 'function') {
+    return redirectDestination({ ctx, AuthUser })
+  }
+
+  return redirectDestination
+}
+
+export const getRedirectToLoginDestination = (
+  authRedirectDestination,
+  ctx,
+  AuthUser
+) => {
+  if (!authRedirectDestination) {
+    throw new Error(
+      'The "authPageURL" config setting must be set when using `REDIRECT_TO_LOGIN`.'
+    )
+  }
+  const destination = getDestination(ctx, AuthUser, authRedirectDestination)
+
+  if (!destination) {
+    throw new Error(
+      'The "authPageURL" must be set to a non-empty string, an object literal containing "url" and "basePath", or resolve to either'
```

Comment:
* The "url" should be "destination" instead.
* Just remove "basePath", as it's optional
* Change "resolve to either" to "a function that returns either"

Reply:
Suggestion: 'The "authPageURL" must be set to a non-empty string, an object containing a "destination" property, or a function that returns either'